lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/service/dva/DvaThirdPartyDocumentGateway.java

```diff
@@ -1,41 +1,320 @@
 
 import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
 import java.security.cert.CertificateException;
+import java.security.spec.InvalidKeySpecException;
 import java.text.ParseException;
+import java.time.LocalDate;
+import java.util.Base64;
+import java.util.List;
+import java.util.Objects;
+
+import static uk.gov.di.ipv.cri.drivingpermit.library.domain.IssuingAuthority.DVA;
+import static uk.gov.di.ipv.cri.drivingpermit.library.metrics.Definitions.ISSUING_AUTHORITY_PREFIX;
+import static uk.gov.di.ipv.cri.drivingpermit.library.metrics.Definitions.THIRD_PARTY_DCS_RESPONSE_OK;
+import static uk.gov.di.ipv.cri.drivingpermit.library.metrics.Definitions.THIRD_PARTY_DCS_RESPONSE_TYPE_ERROR;
+import static uk.gov.di.ipv.cri.drivingpermit.library.metrics.Definitions.THIRD_PARTY_REQUEST_CREATED;
 
 public class DvaThirdPartyDocumentGateway implements ThirdPartyAPIService {
 
     private static final String SERVICE_NAME = DvaThirdPartyDocumentGateway.class.getSimpleName();
-
+    private static final Logger LOGGER = LogManager.getLogger();
     private DvaCryptographyService dvaCryptographyService;
+    private final ObjectMapper objectMapper;
+    private final ConfigurationService configurationService;
+    private final HttpRetryer httpRetryer;
+    private final EventProbe eventProbe;
+    private static final String OPENID_CHECK_METHOD_IDENTIFIER = "data";
+    private static final String IDENTITY_CHECK_POLICY = "published";
 
     public DvaThirdPartyDocumentGateway(
             ObjectMapper objectMapper,
             DvaCryptographyService dvaCryptographyService,
             ConfigurationService configurationService,
             HttpRetryer httpRetryer,
-            EventProbe eventProbe) {}
+            EventProbe eventProbe) {
+        this.objectMapper = objectMapper;
+        this.dvaCryptographyService = dvaCryptographyService;
+        this.configurationService = configurationService;
+        this.httpRetryer = httpRetryer;
+        this.eventProbe = eventProbe;
+    }
 
     @Override
     public String getServiceName() {
         return SERVICE_NAME;
     }
 
     @Override
-    public DocumentCheckResult performDocumentCheck(DrivingPermitForm drivingPermitForm)
+    public DocumentCheckResult performDocumentCheck(DrivingPermitForm drivingPermitData)
             throws InterruptedException, OAuthHttpResponseExceptionWithErrorBody,
-                    CertificateException, ParseException, JOSEException, IOException {
-        return null;
+                    CertificateException, ParseException, JOSEException, IOException,
+                    NoSuchAlgorithmException {
+        LOGGER.info("Mapping person to third party document check request");
+        DvaPayload dvaPayload = new DvaPayload();
+
+        IssuingAuthority issuingAuthority;
```

Comment:
@ElliotMurphyGDS Im wondering if we should move this bit of logic up its the same regardless of the service being used. What do you think?

Reply:
Your right the logic is the same but the field mappings are different per service...could make a shared utility to reduce duplication/simplify the performDocumentCheck methods though I'll have a think

Reply:
But right now were in the DVA document gateway so it will only ever be considering one service and one party and as this logic is just adding a metric it really should sit in the IdentityVerificationService just before perform document check